internal/host/hostcommands/instruction_manager.go

```diff
@@ -55,16 +55,17 @@ type InstructionConfig struct {
 	SupportL2            bool          `envconfig:"SUPPORT_L2" default:"true"`
 	InstallationTimeout  uint          `envconfig:"INSTALLATION_TIMEOUT" default:"0"`
 	DiskCheckTimeout     time.Duration `envconfig:"DISK_CHECK_TIMEOUT" default:"8m"`
+	SupportFreeAddresses bool          `envconfig:"SUPPORT_FREE_ADDRESSES" default:"true"`
 	ReleaseImageMirror   string
 	CheckClusterVersion  bool
 }
 
 func NewInstructionManager(log logrus.FieldLogger, db *gorm.DB, hwValidator hardware.Validator, ocRelease oc.Release,
-	instructionConfig InstructionConfig, connectivityValidator connectivity.Validator, eventsHandler events.Handler, versionHandler versions.Handler) *InstructionManager {
+	instructionConfig InstructionConfig, connectivityValidator connectivity.Validator, eventsHandler events.Handler, versionHandler versions.Handler, enableKubeAPI bool) *InstructionManager {
```

Comment:
instructions manager should not know or care about the kubeAPI.
Instead, we should add an 'enableFreeIPCheck' flag to the InstructionConfig, which main will set

Reply:
OK